src/Support/UrlGenerator/DefaultUrlGenerator.php

```diff
@@ -28,7 +28,13 @@ public function getBaseMediaDirectoryUrl()
 
     public function getPath(): string
     {
-        $pathPrefix = $this->getDisk()->getAdapter()->getPathPrefix();
+        $adapter = $this->getDisk()->getAdapter();
+
+        if (! \method_exists($adapter, 'getPathPrefix')) {
```

Comment:
I think if we'd check on instanceof `CachedAdapter` would be more readable.

Als, drop the `\` before `method_exists`, it's a micro optimisation that we don't do anywhere in our code.